dataplane/linux/int_dataplane.go

```diff
@@ -339,12 +341,10 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 	dp.iptablesFilterTables = append(dp.iptablesFilterTables, filterTableV4)
 	dp.ipSets = append(dp.ipSets, ipSetsV4)
 
-	routeTableV4 := routetable.New(config.RulesConfig.WorkloadIfacePrefixes, 4, false, config.NetlinkTimeout, config.DeviceRouteSourceAddress)
-	dp.routeTables = append(dp.routeTables, routeTableV4)
-
 	if config.RulesConfig.VXLANEnabled {
-		routeTableVXLAN := routetable.New([]string{"vxlan.calico"}, 4, true, config.NetlinkTimeout, config.DeviceRouteSourceAddress)
-		dp.routeTables = append(dp.routeTables, routeTableVXLAN)
+		routeTableVXLAN := routetable.New([]string{"vxlan.calico"}, 4, true, config.NetlinkTimeout, config.DeviceRouteSourceAddress,
+			config.DeviceRouteProtocol, config.RemoveExternalRoutes)
```

Comment:
I don't think we want to drive this based on the config parameters - we _never_ want to remove external routes on eth0, for example.

Reply:
Yes, I'll update this.